pkg/cloud/services/network/subnets.go

```diff
@@ -174,18 +190,14 @@ func (s *Service) reconcileSubnets() error {
 			record.Warnf(s.scope.InfraCluster(), "FailedNoPublicSubnet", "Expected at least 1 public subnet but got 0")
 			return errors.New("expected at least 1 public subnet but got 0")
 		}
-	} else if unmanagedVPC {
-		if len(subnets) < 1 {
-			record.Warnf(s.scope.InfraCluster(), "FailedNoSubnet", "Expected at least 1 subnet but got 0")
-			return errors.New("expected at least 1 subnet but got 0")
-		}
-	}
 
-	// Proceed to create the rest of the subnets that don't have an ID.
-	if !unmanagedVPC {
+		// Proceed to create the rest of the subnets that don't have an ID.
 		for i := range subnets {
 			subnet := &subnets[i]
-			if subnet.ID != "" {
+
+			// If we have a ResourceID (ARN), the resource was already created,
```

Comment:
The field name and description makes me think it's an ID like `subnet-abc123`, but this comment says it's an ARN. Can you fix / clarify the description and/or comments?

And it's unclear what a "temporary one" is. Currently, the PR says that the `subnet.ID` field has two meanings: `subnet-xxx` for bring-your-own, or any other name for the managed VPC use case. Either way, the field value won't change and therefore can't be temporary, if I understand this right.

Reply:
Clarified, was a leftover